apputils/config_validator.py

```diff
@@ -0,0 +1,52 @@
+"""Validator for config files."""
+
+
+class ConfigValidator(object):
+    sections_required = ["task_config", "log_config", "schedule",
+                         "env_literal", "env_expand"]
+    task_config_required = ['task_name', 'image', 'retry', 'read_only']
+    log_config_required = ['task_started', 'task_finished',
+                           'task_retried', 'task_failed']
+    schedule_required = ['minute', 'hour', 'day_of_week',
+                         'day_of_month', 'month_of_year']
+
+    @classmethod
+    def validate_config(cls, config):
+        """Validate required fields are present and module is 'scheduler'."""
+        msg = ""
+        if ("service" not in config
+                or "module" not in config["service"]
+                or config["service"]["module"] != "scheduler"):
+            # We bail if there is no service config, or if service is not
+            # set to "scheduler".
+            msg += ("ConfigValidator: Not a scheduler config!")
+            return False
+        missing_sections = [x for x in cls.sections_required
+                            if x not in config]
+        if missing_sections:
```

Comment:
Can you utilitize https://docs.python.org/2/library/configparser.html instead of using manual check

exception ConfigParser.NoOptionError¶
Exception raised when a specified option is not found in the specified section.